src/vm/codeversion.cpp

```diff
@@ -50,13 +50,20 @@ bool NativeCodeVersion::operator!=(const NativeCodeVersion & rhs) const { return
 #define CORPROF_E_RUNTIME_SUSPEND_REQUIRED 0x80131381
 
 #ifndef DACCESS_COMPILE
-NativeCodeVersionNode::NativeCodeVersionNode(NativeCodeVersionId id, MethodDesc* pMethodDesc, ReJITID parentId) :
+NativeCodeVersionNode::NativeCodeVersionNode(
+    NativeCodeVersionId id,
+    MethodDesc* pMethodDesc,
+    ReJITID parentId,
+    NativeCodeVersion::OptimizationTier optimizationTier)
+    :
     m_pNativeCode(NULL),
     m_pMethodDesc(pMethodDesc),
     m_parentId(parentId),
     m_pNextMethodDescSibling(NULL),
     m_id(id),
-    m_optTier(NativeCodeVersion::OptimizationTier0),
+#ifdef FEATURE_TIERED_COMPILATION
+    m_optTier(optimizationTier),
+#endif
```

Comment:
Nit: I don't think this will compile if `FEATURE_TIERED_COMPILATION` is not defined. Not sure if it is worth fixing though.

Reply:
It looked like `m_optTier` is defined under `FEATURE_TIERED_COMPILATION` so I thought that it would not compile without the define before the change

Reply:
Sorry, should have been more specific -- `OptimizationTier` is only defined under `FEATURE_TIERED_COMPILATION`, so it is the argument list where the problem arises, not the initializer list.

Reply:
Ah I see. I think it's fine how you have it ... varying signatures based on defines is always annoying.